src/main/resources/public/js/app.js

```diff
@@ -53,6 +54,7 @@ function fire(row, column) {
         dataType: "json"
     });
 
+
     request.done(function( currModel ) {
         displayGameState(currModel);
         gameModel = currModel;
```

Comment:
Right after this line is where you should check if the game is over. When you place a shot, this javascript fire function gets called, and it sends a request to the server. When the javascript recieves a response from the server (with the updated game model), it calles the function inside `request.done(...)`.

So if you perform the check inside `request.done`, you can just alert here if the game is over.